packages/client/src/v2-events/layouts/form/FormHeader.tsx

```diff
@@ -41,21 +42,38 @@ const messages = defineMessages({
   }
 })
 
+export const ActionIcons: { [key in ActionType]: React.JSX.Element } = {
+  CREATE: <DeclarationIcon color={getDeclarationIconColor()} />,
+  ASSIGN: <DeclarationIcon color={getDeclarationIconColor()} />,
+  UNASSIGN: <DeclarationIcon color={getDeclarationIconColor()} />,
+  REGISTER: <DeclarationIcon color={getDeclarationIconColor()} />,
+  VALIDATE: <DeclarationIcon color={getDeclarationIconColor()} />,
+  REQUEST_CORRECTION: <DeclarationIcon color={getDeclarationIconColor()} />,
+  REJECT_CORRECTION: <DeclarationIcon color={getDeclarationIconColor()} />,
+  APPROVE_CORRECTION: <DeclarationIcon color={getDeclarationIconColor()} />,
+  DETECT_DUPLICATE: <DeclarationIcon color={getDeclarationIconColor()} />,
+  NOTIFY: <DeclarationIcon color={getDeclarationIconColor()} />,
+  DECLARE: <DeclarationIcon color={getDeclarationIconColor()} />,
+  DELETE: <DeclarationIcon color={getDeclarationIconColor()} />,
+  PRINT_CERTIFICATE: <Print color={getDeclarationIconColor()} />,
+  CUSTOM: <DeclarationIcon color={getDeclarationIconColor()} />
+}
```

Comment:
Would it work to let the actions themselves pass down the icon they want to use from the action module (features/events/actions*) instead of having a centralised configuration? I'm saying this as the actions should be modular. You can imagine one day we might let a country define their own action module as React (or a microfrontend like you suggested) in their country config but for us to be able to do so, we shouldn't create too much dependency on "what the actions are" in the code outside the modules.

Reply:
agree, done